### PR TITLE
perf(code-editor): do not run change detection whenever the editor does re-layout

### PR DESCRIPTION
**Pull request by arturovt**

### CODE REVIEW

File: libs/angular-code-editor/src/lib/code-editor.component.ts [hunk 1]
```diff
@@ -352,32 +362,29 @@ export class TdCodeEditorComponent
...
+        this._heightSubject.asObservable().pipe(distinctUntilChanged())
+      )
+        .pipe(debounceTime(100), takeUntil(this._destroy))
+        .subscribe(() => {
+          // Note: this is being called outside of the Angular zone since we don't have to
+          // run change detection whenever the editor resizes itself.
+          this.layout();
+        })
+    );
+    this._ngZone.runOutsideAngular(() =>
```
Comment (owilliams320) [inline libs/angular-code-editor/src/lib/code-editor.component.ts:378]: not sure this really needs to be run outside of the ngzone? not seeing what would trigger change detection if its just getting values and triggering next in a behavior subject

**Review by owilliams320 — CHANGES_REQUESTED** [review]

File: libs/angular-code-editor/src/lib/code-editor.component.ts (hunk 1 above)
Comment (arturovt) [reply libs/angular-code-editor/src/lib/code-editor.component.ts:378]: There's _timer_ (that acts as a setInterval) that triggers change detection

**Comment by arturovt** [comment]
@owilliams320 friendly ping

**Review by owilliams320 — APPROVED** [review]

**Pull request merged**